pandas/tests/indexes/interval/test_interval.py

```diff
@@ -926,6 +926,57 @@ def test_pickle_round_trip_closed(self, closed):
         tm.assert_index_equal(result, idx)
 
 
+@pytest.mark.parametrize(
+    "test_case",
+    [
+        (
+            "float64",
+            IntervalIndex.from_breaks([0.0, 0.5, 1.0]),
+            ["(0.0, 0.5]", "(0.5, 1.0]"],
+        ),
+        ("int64", IntervalIndex.from_breaks([0, 5, 10]), ["(0, 5]", "(5, 10]"]),
+        (
+            "datetime64[ns]",
+            IntervalIndex.from_breaks(
+                [Timestamp(2015, 7, 1), Timestamp(2016, 8, 1), Timestamp(2018, 9, 1)]
+            ),
+            ["(2015-07-01, 2016-08-01]", "(2016-08-01, 2018-09-01]"],
+        ),
+    ],
+)
+def test_from_strings(test_case):
+    """Test the IntervalIndex.from_strings class method."""
+    # See https://github.com/pandas-dev/pandas/pull/41451
+    dtype, expected, string = test_case
+
+    # Attempt to parse the type dynamically
+    parsed_index = IntervalIndex.from_strings(string)
+    assert np.array_equal(parsed_index, expected)
```

Comment:
pls use `tm.assert_index_equal` and construct the expected interval index directly.

Reply:
same for all comparisons.

Reply:
Added in a8cc3b2caf5d3d93030eccf6a8c3759b804817ca